docs/helmrelease-guide/rollbacks.md

```diff
@@ -0,0 +1,80 @@
+# Rollbacks
+
+From time to time a release made by the Helm Operator may fail, this section
+of the guide will explain how you can recover from a failed release by enabling
+rollbacks.
+
+!!! caution
+    Rollbacks of Helm charts containing `StatefulSet` resources can be a
+    tricky operation, and are one of the main reasons automated rollbacks are not
+    enabled by default. Verify a manual rollback (using `helm`) of your Helm
+    chart does not cause any problems before enabling it.
+
+## Enabling rollbacks
+
+When rollbacks for a `HelmRelease` are enabled, the Helm Operator will detect
+a faulty upgrade and instruct Helm to perform a rollback, it will not attempt
+a new upgrade unless it detects a change in values and/or the chart, or [retries
+have been enabled](#enabling-retries-of-rolled-back-releases). Changes are
+detected by comparing the failed release to a fresh dry-run release.
+
+Rollbacks can be enabled by setting `.rollback.enable`:
+
+```yaml
+spec:
+  rollback:
+    enable: true
+```
+
+## Tweaking the rollback configuration
+
+To get more fine-grained control over how the rollback is performed by Helm,
+the `.rollback` of the `HelmRelease` resources offers a couple of additional
+settings.
+
+```yaml
+spec:
+  rollback:
+    enable: true
+    disableHooks: false
+    force: false
+    recreate: false
+    timeout: 300
+```
+
+The definition of the listed keys is as follows:
+
+* `enable`: Enables the performance of a rollback when a release fails.
+* `disableHooks` _(Optional)_: When set to `true`, prevent hooks from running
+  during rollback. Defaults to `false` when omitted.
+* `force` _(Optional)_: When set to `true`, force resource update through
+  delete/recreate if needed. Defaults to `false` when omitted.
+* `recreate` _(Optional)_: When set to `true`, performs pods restart for the
+  resource if applicable. Defaults to `false` when omitted.
+* `timeout` _(Optional)_: Time to wait for any individual Kubernetes operation
```

Comment:
If not mistaken, this is the timeout for the rollback operation, not the duration after which the operator triggers a rollback. Can you please make it clear people should use `.spec.timeout` for that purpose?

Reply:
I was under the impression that this would be clear by it being a nested key under `rollback`.

Would

> Time to wait for any individual Kubernetes operation that is part of the rollback

be more clear to you?

Reply:
Yes, clearer. My point is that it should not be mistaken for `.spec.timeout`, which is mostly what users care about I believe.

Reply:
Another thing that comes to mind now is that people will likely want to increase the `terminationGracePeriod` on the Helm Operator pod when they set this to a high value to prevent releases from getting stuck in a faulty state.

Thanks for the suggestion!